MbDotNet/MountebankRequestProxy.cs

```diff
@@ -121,6 +123,24 @@ private HttpResponseMessage ExecutePost(string resource, string json)
             }
         }
 
+        private void HandleDynamicPort(HttpResponseMessage response, Imposter imposter)
+        {
+            if (imposter.Port == null)
+            {
+                try
+                {
+                    var content = response.Content.ReadAsStringAsync().Result;
+                    Match match = new Regex(".*\"port\": (\\d+),", RegexOptions.Multiline).Match(content);

```

Comment:
I think I would lean towards adding the new model. That way we can let the JSON deserializer do the work for us instead of parsing the content. Make the type internal though so it is not exposed in the package. I would be fine with just putting that class in this file even since this is the only place it will be used.